src/contracts/libraries/SafetyBits.sol

```diff
@@ -73,21 +71,30 @@ library SafetyBits {
         return self;
     }
 
-    function holdUserLock(EscrowKey memory self, address addressPointer) internal pure returns (EscrowKey memory) {
+    function holdUserLock(EscrowKey memory self, address userOpDapp) internal pure returns (EscrowKey memory) {
         self.lockState = _LOCKED_X_USER_X_UNSET;
-        self.addressPointer = addressPointer;
+        self.addressPointer = userOpDapp;
         unchecked {
             ++self.callIndex;
         }
         return self;
     }
 
+    function holdPreSolverLock(EscrowKey memory self) internal pure returns (EscrowKey memory) {
+        self.lockState = _LOCKED_X_PRESOLVERS_X_REQUESTED;
+        self.addressPointer = self.executionEnvironment;
+        return self;
+    }
+
     function holdSolverLock(EscrowKey memory self, address nextSolver) internal pure returns (EscrowKey memory) {
```

Comment:
`holdSolverLock()` is not called anywhere. Should it be called in `_trySolverLock()`?

Reply:
I believe we can delete it since we don't pass this data to the solver. Previously, it was used to pass data to the EE, which would then call the solver.

Reply:
82c8be0